Library/Homebrew/cmd/shellenv.sh

```diff
@@ -3,36 +3,48 @@
 #:  Print export statements. When run in a shell, this installation of Homebrew will be added to your `PATH`, `MANPATH`, and `INFOPATH`.
 #:
 #:  The variables `HOMEBREW_PREFIX`, `HOMEBREW_CELLAR` and `HOMEBREW_REPOSITORY` are also exported to avoid querying them multiple times.
+#:  The variables and `HOMEBREW_SHELLENV_PREFIX` and `HOMEBREW_SHELLENV_SET` will be exported to avoid adding duplicate entries to the environment variables.
```

Comment:
Sorry, poor communication on my side: I think we can remove the `HOMEBREW_SHELLENV_SET` variable entirely as `HOMEBREW_SHELLENV_PREFIX` now handles those cases sufficiently if you use the globs.